src/lib/dotrenderer.ml

```diff
@@ -63,28 +100,41 @@ let parse_markdown (markdown : string) : Block.t list =
           |> List.map String.trim |> String.concat ~sep:"\n"
         in
         (match Block.of_info_string ~default ~body info_str with
-        | Some b -> blocks := b :: !blocks
+        | Some b -> cuirrent_block_list := b :: !cuirrent_block_list
         | None -> ());
         `Default
     | _ -> `Default
   in
 
   let mapper = Cmarkit.Mapper.make ~block () in
   ignore (Cmarkit.Mapper.map_doc mapper doc);
```

Comment:
Looks like what you want here is a `Cmarkit.Folder` instead of a mapper + mutable references?

Reply:
That's much neater now, thanks.